### ISSUE TITLE
Volpiano search results mishandle missing music

### ISSUE THREAD

**Issue by dchiller** [DDMAL/cantus#656]
In the process of looking into #647, I uncovered another issue with volpiano search, which seems to be unrelated. 

When displaying volpiano search results, Cantus Ultimus aligns the standardized text with the volpiano melody. 

So, for example, this chant: 
<img width="594" alt="image" src="https://user-images.githubusercontent.com/11023634/206301269-cf15904e-719c-4ed3-8979-b0c33256b693.png">
shows up like this: 

<img width="542" alt="image" src="https://user-images.githubusercontent.com/11023634/206301298-7c15c4cd-51e4-47e1-b48c-3e41fbf78285.png">

**Comment by annamorphism** [comment]
I've wondered to myself before whether and when the standardized text was going to cause us difficulties...
I can't actually get your "Domine deus" example to show up in a Volpiano search, but I could be doing something wrong on my end (or is this in fact the problem you are looking into?).

Pros: There will definitely be other cases where the Volpiano is matched to the MS text and not the (significantly different) "standard" text e.g. manuscripts with lots of missing pages, so it might be nice to avoid these kinds of gaps. Plus, the Volpiano melodies in the actual sources match the manuscript readings, so it would be sort of pleasing to have them match the search made to get there. 

Cons: This might introduce all kinds of interesting spelling variants to the text search field, which currently takes in the standard spellings (no results for "ceci vident"). And my expectation for the search snippet isn't really to have a complete text to music matchup anyway--I can head to the actual manuscript to see it.  

I'll think on it. I don't really see a strong reason for keeping the "standard spelling" if it's not too hard to switch to the "manuscript spelling," but maybe there was some compelling reason it was done that way in the first instance.

**Comment by dchiller** [comment]
Something (a few things?) are definitely wonky with volpiano search, so I feel pretty safe in saying that it's not a you problem that you can't find "Domine deus" ~~ hopefully not for long though. 

Thanks for thinking this through...we could always run the text search against the standardized text field, but then show the manuscript text with the results....but this may make things more confusing for the user! 

I don't know if Ich or anyone else would have a point of view. Would welcome their thoughts too.

**Comment by dchiller** [comment]
Ok, I have more insight now into the cause of this issue. Basically, right now, when dealing with chants of missing music, Cantus Ultimus looks for curly brackets in the chant text to figure out how to align text and music. Makes sense. However, it looks for curly brackets in the standardized text for the chant, but I imagine we would expect those to exist a lot less frequently that than in the manuscript spelling text.

A quick search of two manuscripts reveals this assumption to be true. I imagine that we only have missing words in standardized texts when the words are missing from all sources, whereas much more often we have missing text in one manuscript but we know what the whole standard text is because we can check against other sources.

**Comment by annamorphism** [comment]
There shouldn't be any curly brackets in the standardized spelling, actually--the practice is to indicate that only in the MS spelling field.  
Missing text is different in the two fields, too--the missing, expected text is in square brackets in the standardized spelling field, but replaced with # in the MS spelling. 
So, if there are going to be manuscripts with a lot of missing parts--holes, missing pages, etc--it will be a *lot* easier to match them to the MS spelling text than to the standardized one.

**Comment by dchiller** [comment]
For your reference, I noticed these two instances of curly brackets in the standardized spelling field: 
- https://cantus.uwaterloo.ca/chant/398391
- https://cantus.uwaterloo.ca/chant/398428

Maybe those are entry errors?

Speaking of these chants, is there a difference in the manuscript spelling between the hash sign (#) and the {xxx} ?

**Comment by dchiller** [comment]
Regardless, to your point it seems like this is another argument for using the MS spelling.

**Comment by annamorphism** [comment]
Good catch! I'm guessing the protocol had not been fully standardized when this source was entered--I'll make a note to clean up the curly brackets.
As for the hash sign, this is the current protocol for missing text, while xxx seems to be some individual data-enterers solution (there are about five "xxx" chants in the database but lots of hash signs.)
